msrezaie/city-scrapers-bismarck
# Build spider: Bismarck Public Schools

## What's this PR do?
This PR adds a new spider for Bismarck Public Schools, Bismarck, ND.

## Why are we doing this?
Built the spider as per request.

## Steps to manually test

1. Ensure the project is installed:
```
pipenv sync --dev
```
2. Activate the virtual env and enter the pipenv shell:
```
pipenv shell
```
3. Run the spider:
```
scrapy crawl bisnd_bps -O test_output.csv
```
4. Monitor the output and ensure no errors are raised.

5. Inspect `test_output.csv` to ensure the data looks valid.

## Are there any smells or added technical debt to note?
Bismarck Public Schools have meetings of different types; of those types, meetings of type 'Regular' has consistent meeting location and start time. However for meetings that are of type not 'Regular', its location and start time is varied. These values for meetings can be found in their agenda document. Though I was not able to access the meeting's agenda document due to 2 reasons:

1. The agenda documents are available only through a SharePoint webpage in docx format.
2. I was not able to access and scrape the webpage with scrapy due to a recurring redirect measure set by the organization's server. 

Currently the meeting's location and start time is hardcoded, and is taken from the organization's following webpage: https://www.bismarckschools.org/Page/398. Also, an instruction note has been added to the `time_notes` field on where to look for the details of the location and start time of meetings.
